FireboltNETSDK/Client/FireboltParameterCollection.cs

```diff
@@ -29,158 +34,80 @@ public sealed class FireboltParameterCollection : DbParameterCollection, IList<F
         /// <inheritdoc/>
         public override int Add(object value)
         {
-            if (value == null)
-                throw new ArgumentNullException(nameof(value));
-
-            var parameter = (FireboltParameter)value;
-            return Add(parameter);
-        }
-
-        /// <summary>
-        /// Creates, adds to the collection and returns a new parameter with specified name and value.
-        /// </summary>
-        /// <param name="parameterName">The name of the parameter.</param>
-        /// <param name="value">The value of the paramter.</param>
-        /// <returns>A new <see cref="FireboltParameter"/> added to the collection.</returns>
-        public FireboltParameter AddWithValue(string parameterName, object? value)
-        {
-            var parameter = new FireboltParameter(parameterName) { Value = value };
+            DbParameter parameter = typeof(DbParameter).IsAssignableFrom(value.GetType()) ? (DbParameter)value : new FireboltParameter(FireboltParameter.defaultParameterName, value);
             Add(parameter);
-            return parameter;
+            return _parameters.Count - 1;
```

Comment:
Why do we do `-1` here?

Reply:
Because according to the documentation of this method declared in `DbParameterCollection` it returns "The index of the System.Data.Common.DbParameter object in the collection."
The index is zero based, that means that we should subtract 1 from the collection size.